awscli/autoprompt/prompttoolkit.py

```diff
@@ -55,7 +57,7 @@ class PromptToolkitPrompter:
 
     """
     def __init__(self, completion_source, driver, completer=None,
-                 factory=None, app=None):
+                 factory=None, app=None, index_filename=None):
```

Comment:
Instead of accepting the `index_filename`, it probably makes a bit more sense to accept a `cli_parser` and then create one if `None` is specified. Mainly, this class should not really know that a model index is used rather it only really cares about having a parser that it can use to parse the results from the input buffer and that could use whatever implementation.